### PR TITLE
feat: migrate to 0.10.0

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]
I'll review tests later

File: README.md [hunk 1]
```diff
@@ -777,17 +777,17 @@ wait_agent(3, action, ScKeynodes[ActionStatus.ACTION_FINISHED])
...
 ```python
-def create_action(*concepts: Idtf) -> ScAddr: ...
+def generate_action(*concepts: Idtf) -> ScAddr: ...
 ```
 
-Function that creates arguments of action
+Function that generates arguments of action
 
 ```python
 def add_action_arguments(action_node: ScAddr, arguments: Dict[ScAddr, IsDynamic]) -> None: ...
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply README.md:793]: (

File: tests/test_sc_sets/test_sc_set.py [hunk 2]
```diff
@@ -4,47 +4,47 @@
...
 
-    def test_create_without_set_node(self):
-        element1 = create_node(sc_types.NODE_CONST)
-        element2 = create_node(sc_types.NODE_CONST)
+    def test_generate_without_set_node(self):
+        element1 = generate_node(sc_type.CONST_NODE)
+        element2 = generate_node(sc_type.CONST_NODE)
         sc_set = ScSet(element1, element2)
-        self.assertEqual(check_elements(sc_set.set_node)[0], sc_types.NODE_CONST)
+        self.assertEqual(check_elements(sc_set.set_node)[0], sc_type.CONST_NODE)
```
Comment (kilativ-dotcom) [inline tests/test_sc_sets/test_sc_set.py:28]:
```suggestion
        self.assertEqual(get_elements_types(sc_set.set_node)[0], sc_type.CONST_NODE)
```

Comment (kilativ-dotcom) [reply tests/test_sc_sets/test_sc_set.py:28]: fix it everywhere

File: tests/test_classes.py
Comment (kilativ-dotcom) [inline tests/test_classes.py:1]: I am getting this error when running tests
image

File: tests/test_utils/test_common_utils.py [hunk 3]
```diff
@@ -119,23 +119,20 @@ def test_relation_utils(self):
...
 
     def test_deletion_utils(self):
-        src, rrel_trg, nrel_trg = create_nodes(sc_types.NODE_CONST, sc_types.NODE_CONST, sc_types.NODE_CONST)
-        rrel_edge = create_binary_relation(sc_types.EDGE_ACCESS_CONST_POS_PERM, src, rrel_trg)
-        nrel_edge = create_norole_relation(src, nrel_trg)
-        assert delete_edges(src, rrel_trg, sc_types.EDGE_ACCESS_VAR_POS_PERM)
-        assert delete_elements(nrel_edge, src, rrel_trg, nrel_trg)
-
-        result = client.check_elements(rrel_edge)[0]
-        assert result.is_valid() is False
```
Comment (kilativ-dotcom) [inline tests/test_utils/test_common_utils.py:141]: why did you remove these 2 lines?

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: tests/test_utils/test_common_utils.py (hunk 3 above)
Comment (NikitaZotov) [reply tests/test_utils/test_common_utils.py:141]: it throw exception, because element is not valid

File: tests/test_classes.py
Comment (NikitaZotov) [reply tests/test_classes.py:1]: you don't launch sc-server

File: src/sc_kpm/utils/common_utils.py [hunk 4]
```diff
@@ -4,147 +4,284 @@
...
-def get_element_by_norole_relation(src: ScAddr, nrel_node: ScAddr) -> ScAddr:
-    search_result = search_norole_relation_template(src, nrel_node)
+def search_element_by_non_role_relation(src: ScAddr, nrel_node: ScAddr) -> ScAddr:
+    search_result = search_non_role_relation_template(src, nrel_node)
     return search_result.get(ScAlias.ELEMENT) if search_result else ScAddr(0)
 
 
+def get_element_by_norole_relation(src: ScAddr, nrel_node: ScAddr) -> ScAddr:
+    warnings.warn(
+        "Common utils 'get_element_by_norole_relation' method is deprecated. Use `search_element_by_non_role_relation` method instead.",
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply src/sc_kpm/utils/common_utils.py:267]: no, black says that all is good

File: src/sc_kpm/utils/common_utils.py [hunk 5]
```diff
@@ -4,147 +4,284 @@
...
+
+
+def search_non_role_relation_template(src: ScAddr, nrel_node: ScAddr) -> Optional[ScTemplateResult]:
+    return _search_relation_template(src, nrel_node, sc_type.VAR_COMMON_ARC)
 
 
 def search_norole_relation_template(src: ScAddr, nrel_node: ScAddr) -> Optional[ScTemplateResult]:
-    return _search_relation_template(src, nrel_node, sc_types.EDGE_D_COMMON_VAR)
+    warnings.warn(
+        "Common utils 'search_norole_relation_template' method is deprecated. Use `search_non_role_relation_template` method instead.",
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply src/sc_kpm/utils/common_utils.py:249]: no, black says that all is good

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -22,10 +23,13 @@ jobs:
...
         pip install -U wheel
         pip install -U setuptools
         python -m pip install -U pip
-    - run: pip install tox
-#    - run: tox -e ${{ matrix.tox }}
+    - name: Install tox
+      run: pip install tox
+
+    #- name: Run tests
+    #  run: tox -e ${{ matrix.tox }}
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply .github/workflows/tests.yml:35]: I will create issue

File: README.md [hunk 7]
```diff
@@ -599,30 +598,27 @@ def get_link_content(link: ScAddr) -> Union[str, int]: ...
...
+from sc_client.constants import sc_type
 from sc_kpm import ScKeynodes
-from sc_kpm.utils import create_node
-from sc_kpm.utils import get_system_idtf
+from sc_kpm.utils import get_element_system_identifier
 
-lang_en = create_node(sc_types.NODE_CONST_CLASS)  # ScAddr(...)
-idtf = get_system_idtf(lang_en)  # "lang_en"
-assert ScKeynodes[idtf] == lang_en
+idtf = get_element_system_identifier(lang_en)  # "lang_en"
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply README.md:621]: where?

File: src/sc_kpm/utils/__init__.py [hunk 8]
```diff
@@ -17,11 +18,25 @@
...
+    get_connector,
+    get_connectors,
     get_edge,
     get_edges,
     get_element_by_norole_relation,
     get_element_by_role_relation,
+    get_element_system_identifier,
     get_link_content_data,
     get_system_idtf,
+    search_element_by_non_role_relation,
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply src/sc_kpm/utils/__init__.py:40]: What are they? There's a lot of them.

[the remark this answers is not in the document]
Comment (kilativ-dotcom) [reply src/sc_kpm/utils/__init__.py:40]: like get_connector, get_connectors, get_element_by_role_relation

File: tests/test_classes.py
Comment (kilativ-dotcom) [reply tests/test_classes.py:1]: I run sc-machine in another terminal
also now some tests don't pass
image
FAILED tests/test_keynodes.py::KeynodesTests::test_rrel - AssertionError: False is not true
FAILED tests/test_sc_sets/test_sc_numbered_set.py::ScNumberedSetTestCase::test_generate_large - AssertionError: Lists differ: [] != [ScAddr(196736), ScAddr(196737), ScAddr(196738)]
FAILED tests/test_sc_sets/test_sc_numbered_set.py::ScNumberedSetTestCase::test_get_elements - AssertionError: Lists differ: [] != [ScAddr(196775), ScAddr(196776), ScAddr(196778)]
FAILED tests/test_sc_sets/test_sc_numbered_set.py::ScNumberedSetTestCase::test_remove - AssertionError: 0 != 2

File: src/sc_kpm/utils/common_utils.py [hunk 9]
```diff
@@ -4,147 +4,284 @@
...
     content_type: ScLinkContentType = ScLinkContentType.STRING,
-    link_type: ScType = sc_types.LINK_CONST,
+    link_type: ScType = sc_type.CONST_NODE_LINK,
 ) -> ScAddr:
-    return create_links(content, content_type=content_type, link_type=link_type)[0]
+    warnings.warn(
+        "Common utils 'create_link' method is deprecated. Use `generate_link` method instead.",
+        DeprecationWarning,
+    )
+    return generate_link(content, content_type=content_type, link_type=link_type)[0]
```
[the remark this answers is not in the document]
Comment (kilativ-dotcom) [reply src/sc_kpm/utils/common_utils.py:88]: you forgot to push this

File: src/sc_kpm/sc_keynodes.py [hunk 10]
```diff
@@ -54,16 +63,23 @@ def resolve(cls, identifier: Idtf, sc_type: Optional[ScType]) -> ScAddr:
...
 
     def rrel_index(cls, index: int) -> ScAddr:
-        """Get rrel_i node. Max rrel index is 10"""
+        """Get rrel_i node. Max rrel index is 10. Min rrel is 1."""
         if not isinstance(index, int):
             raise TypeError("Index of rrel node must be int")
         if index > cls._max_rrel_index:
             raise KeyError(f"You cannot use rrel more than {cls._max_rrel_index}")
-        return cls.resolve(f"rrel_{index}", NODE_CONST_ROLE)  # pylint: disable=no-value-for-parameter
+        if index < 1:
```
[the remark this answers is not in the document]
Comment (kilativ-dotcom) [reply src/sc_kpm/sc_keynodes.py:80]: you forgot to push this

File: src/sc_kpm/utils/common_utils.py [hunk 11]
```diff
@@ -4,147 +4,284 @@
...
+
 def _search_relation_template(src: ScAddr, rel_node: ScAddr, rel_type: ScType) -> Optional[ScTemplateResult]:
     template = ScTemplate()
-    template.triple_with_relation(
+    template.quintuple(
         src,
-        rel_type >> ScAlias.ACCESS_EDGE,
-        sc_types.UNKNOWN >> ScAlias.ELEMENT,
-        sc_types.EDGE_ACCESS_VAR_POS_PERM,
+        rel_type >> ScAlias.MEMBERSHIP_ARC,
```
[the remark this answers is not in the document]
Comment (kilativ-dotcom) [reply src/sc_kpm/utils/common_utils.py:230]: you forgot to push this

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]
not all was fixed and tests are failing

File: tests/test_classes.py
Comment (NikitaZotov) [reply tests/test_classes.py:1]: image

Comment (kilativ-dotcom) [reply tests/test_classes.py:1]: as it turned out some tests are failing when kb.bin is empty

File: tests/test_utils/test_common_utils.py (hunk 3 above)
Comment (kilativ-dotcom) [reply tests/test_utils/test_common_utils.py:141]: but previously unknown type was returned for not valid elements

File: src/sc_kpm/sc_keynodes.py [hunk 12]
```diff
@@ -54,16 +63,23 @@ def resolve(cls, identifier: Idtf, sc_type: Optional[ScType]) -> ScAddr:
...
     def rrel_index(cls, index: int) -> ScAddr:
-        """Get rrel_i node. Max rrel index is 10"""
+        """Get rrel_i node. Max rrel index is 10. Min rrel is 1."""
         if not isinstance(index, int):
             raise TypeError("Index of rrel node must be int")
         if index > cls._max_rrel_index:
             raise KeyError(f"You cannot use rrel more than {cls._max_rrel_index}")
-        return cls.resolve(f"rrel_{index}", NODE_CONST_ROLE)  # pylint: disable=no-value-for-parameter
+        if index < cls._min_rrel_index:
+            raise KeyError(f"You cannot use rrel less than {cls._min_rrel_index}")
```
Comment (kilativ-dotcom) [inline src/sc_kpm/sc_keynodes.py:81]: there is test for max boundary but no test for min boundary

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply README.md:789]: It is python feature

**Review by kilativ-dotcom — APPROVED** [review]
